src/user.cc

```diff
@@ -180,15 +182,22 @@ TimeEntry *User::Start(
     te->SetTID(task_id);
     te->SetTags(tags);
 
-    if (!duration.empty()) {
-        int seconds = Formatter::ParseDurationString(duration);
-        te->SetDurationInSeconds(seconds);
-        te->SetStop(now);
-        te->SetStart(te->Stop() - te->DurationInSeconds());
+    if (started == 0 && ended == 0) {
+        if (!duration.empty()) {
+            int seconds = Formatter::ParseDurationString(duration);
+            te->SetDurationInSeconds(seconds);
+            te->SetStop(now);
+            te->SetStart(te->Stop() - te->DurationInSeconds());
+        } else {
+            te->SetDurationInSeconds(-now);
+            // dont set Stop, TE is running
+            te->SetStart(now);
+        }
     } else {
-        te->SetDurationInSeconds(-now);
-        // dont set Stop, TE is running
-        te->SetStart(now);
+        int seconds = int(ended - started);
```

Comment:
Introduce the `started` and `ended` to able to create new empty TE at particular time.